lib/trento/software_updates/discovery.ex

```diff
@@ -36,24 +37,14 @@ defmodule Trento.SoftwareUpdates.Discovery do
   def get_upgradable_packages(system_id),
     do: adapter().get_upgradable_packages(system_id)
 
-  @spec discover_software_updates :: {:ok, {list(), list()}}
+  @spec discover_software_updates :: :ok
   def discover_software_updates do
-    {:ok,
-     Hosts.get_all_hosts()
-     |> Enum.map(fn
-       %HostReadModel{id: host_id, fully_qualified_domain_name: fully_qualified_domain_name} ->
-         case discover_host_software_updates(host_id, fully_qualified_domain_name) do
-           {:error, error} ->
-             {:error, host_id, error}
-
-           {:ok, _, _, _} = success ->
-             success
-         end
-     end)
-     |> Enum.split_with(fn
-       {:ok, _, _, _} -> true
-       _ -> false
-     end)}
+    Enum.each(Hosts.get_all_hosts(), fn
+      %HostReadModel{id: host_id} ->
```

Comment:
Do we need to go through a command/aggregate on this?
I mean, it already was doing what we need.
I understand that this change is done to have the `SoftwareUpdatesDiscoveryRequested` event, but honestly, I think that is better not to have these kind of events in user interactions. We discarded this for wanda for example. At the end, the event is not changing anything in the aggregate, and we are adding extra steps to the process for something that already works fine.

Reply:
To me the process is 
`DiscoverSoftwareUpdates` -> `SoftwareUpdatesDiscoveryRequested` 
-> event handler -> `CompleteSoftwareUpdatesDiscovery` (or `UpdateSoftwareUpdatesDiscoveryHealth`) 
-> `SoftwareUpdatesDiscoveryCompleted` (or `SoftwareUpdatesHealthUpdated`/`Changed`)
regardless of where it is triggered. 
It is just consistent. Also in the history of events.

> the event is not changing anything in the aggregate
>
correct, for now we do not change any state.
However we're about to change `software_updates_discovery_health` to support also `not_set` and we might also have the opportunity to set the state to something like `running` maybe and have that information broadcasted to interested parties (the frontend for example).

Besides this the code is much simplified.

The main rationale is consistency in the process and simplification of that piece of code.
The extra opportunity for the `running` state is just a nice side effect.
What do you think?

Reply:
For me it just makes thinks more complex. Instead of a direct path (1 function), we use a function, command, event, event listener, function. (from command on, multiplied by the number of hosts).

All of this, just to add a event in the store, which actually I would prefer not to have it.

Imagine that you have 100 hosts, and you start saving SUMA settings from the frontend. Every single change emits 100 events. And if you fail to put the correct data in the form (which we allow), more.
For a thing, that is not a real event (from my point of view), it is just a user intervention.
More over, we had the same discussion for wanda, and decided not to add such events.

We don't necessarily need to go through the same process to reach the same goal, discover udpates.

Reply:
the problem with many events hosts is that we'd need to do many requests to suma, sequentially.

Yes we could add whatever library do do concurrent things, but we'd get that concurrency for free (or almost free with an extra event in the history) by just leveraging current architecture.

Reply:
but still, I'll revert the change and see what happens